app/grandchallenge/participants/renderers.py

```diff
@@ -0,0 +1,32 @@
+import json
+
+from rest_framework.settings import api_settings
+from rest_framework.utils.encoders import JSONEncoder
+from rest_framework_csv.renderers import PaginatedCSVRenderer
+
+
+class RegistrationRequestCSVRenderer(PaginatedCSVRenderer):
```

Comment:
Rather than flattening stuff with a special renderer why not serialize the answers instead?

Reply:
It was 50/50. I had the answers as a `Charfield` initially, but it felt really strange to have a JSON string in a JSON object for the regular API.

Reply:
That's what it is though - it is a JSON field so in the future may not necessarily be a string. I think it is best to take the same approach as the evaluation serialisation and serialise things on the answer level, rather than the registration request. Permissions are already implemented for that so is less impact. You would then get some duplicate data that is serialised along with each answer (e.g. the username, the request status, the question text) but that is okay for now, it's not much.

Reply:
Do you mean to have a CSV row per answer? 

If so, what about registrees that did not answer any questions? IMHO they should be included in any challenge registration overview. Plus, there are no permissions for the `RegistrationQuestionAnswer`s.

> Permissions are already implemented for that so is less impact.

Are you referring to the permissions for the  `RegistrationQuestionAnswer`s? Those were deemed to be overkill and were removed from the original implementation.

Reply:
Empty answers are saved for users that don't answer, right? The permissions weren't added because they weren't needed. Now they are needed if you want to add this view in line with the existing framework. We should do that rather than trying to fight it with custom renderers specific to one particular field.

Reply:
Empty answers are indeed saved (by design), but if there are no questions at the time of sign-up, the registree would now show up in an answer-level serialization.

Reply:
Then add another viewset, serialiser and permissions for the registration requests and link to those from the answers.

Reply:
Ok. The current plan (somewhere in the next cooldown week) is to have two routes:

- Link for downloading the registration requests.
- Link for downloading the answers to question registrations (with a crosslink to the registration requests).